### PR TITLE
MAUI-1269: Improve performance of Survey Response page in Admin Panel

### PR DESCRIPTION
**Pull request by rohan-bes**
### Issue MAUI-1269:

This page runs really slowly, and when a user navigates to it it hangs the UI for 10+ seconds.

Some basic improvements can get it loading fairly smoothly.
1. Add a debounce to the `DataFetchTable` so that it waits between inputs, currently we send an api request for every keystroke
2. Add a few indexes on the `survey_response_table`
3. Re-work how the `count()` works in the route so that it only joins the reference tables if needed

### CODE REVIEW

File: packages/central-server/src/apiV2/surveyResponses/GETSurveyResponses.js [hunk 1]
```diff
@@ -48,4 +49,16 @@ export class GETSurveyResponses extends GETHandler {
     // Apply regular permissions
     return this.getPermissionsFilter(dbConditions, options);
   }
+
+  async countRecords(criteria) {
+    // Only join tables that we are filtering on
+    const { multiJoin } = getQueryOptionsForColumns(
```
Comment (rohan-bes) [inline packages/central-server/src/apiV2/surveyResponses/GETSurveyResponses.js:55]: The base `countRecords()` joins all the tables required in the columns. However joining on the `entity` and `survey` tables takes the query time up to ~8 seconds (compared to < 1 second without joining).

It does mean however that this number _could_ be higher than the actual row count returned by this route, as it's possible there are survey responses with no reference entities. We could put a constraint against that in the database? Or we could just leave this as possibly saying there are more records than will appear in the Admin Panel (which isn't going to break anything)

File: packages/database/src/migrations/20220920062403-AddEntityIdIndexOnSurveyResponseTable-modifies-schema.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+  type = dbm.dataType;
+  seed = seedLink;
+};
+
+exports.up = async function (db) {
+  await db.runSql(`CREATE INDEX survey_response_entity_id_idx ON survey_response (entity_id);`);
+  await db.runSql(`CREATE INDEX survey_response_outdated_id_idx ON survey_response (outdated);`);
+  await db.runSql(
+    `CREATE INDEX survey_response_data_time_idx ON survey_response (data_time DESC);`,
+  );
```
Comment (rohan-bes) [inline packages/database/src/migrations/20220920062403-AddEntityIdIndexOnSurveyResponseTable-modifies-schema.js:22]: We're kinda building up an 'index per column` pattern on the `survey_response` table... I don't know if that's a database anti-pattern or not... but it's hard to know how we will want to query the table...

Comment (billli0) [reply packages/database/src/migrations/20220920062403-AddEntityIdIndexOnSurveyResponseTable-modifies-schema.js:22]: Emm and even we have `user_id` index, it still takes some times to load in admin panel when searching by user name, that could relates to the join table though.

**Review by billli0 — APPROVED** [review]
Looks good! Thanks for the boost up, that makes the UI smoother.

File: packages/central-server/src/apiV2/surveyResponses/GETSurveyResponses.js (hunk 1 above)
Comment (billli0) [reply packages/central-server/src/apiV2/surveyResponses/GETSurveyResponses.js:64]: No reference entities means no entity name showing up in admin panel?

Comment (rohan-bes) [reply packages/central-server/src/apiV2/surveyResponses/GETSurveyResponses.js:64]: No, this is just the `count` part, so it basically means that we'll count survey responses as existing even if they have no related entity 👍

Comment (billli0) [reply packages/central-server/src/apiV2/surveyResponses/GETSurveyResponses.js:64]: Ah cool, that should be ok then.

**Pull request merged**